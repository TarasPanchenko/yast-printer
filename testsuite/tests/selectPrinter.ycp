/**
 * File:
 *  selectPrinter.ycp
 *
 * Module:
 *  Printer configurator
 *
 * Summary:
 *  Tests for basic translation functions.
 *
 * Authors:
 *  Petr Blahos <blahos@suse.cz>, 2001
 *
 * $Id$
 *
 */

{
    import "Pkg";

    include "testsuite.ycp";
    map I_READ = $[
	"target" : $[
	    "size" : -1,
	    "tmpdir" : "/tmp/",
	    "yast2" : nil,
	],
	"sysconfig" : $[
	    "console" : $[
		"CONSOLE_FONT" : "",
		"CONSOLE_SCREENMAP" : "",
		"CONSOLE_UNICODEMAP" : "",
		"CONSOLE_MAGIC" : "",
		"CONSOLE_ENCODING" : "",
	    ],
	    "language" : $[
		"RC_LANG" : "",
		"DEFAULT_LANGUAGE" : "",
	    ],
	],
	"etc" : $[
	    "install_inf" : $[
		"Cmdline" : "",
		"InstMode" : nil,
	    ],
	],
	"probe" : $[
	    "architecture" : "i386",
	    "has_apm" : true,
	    "has_pcmcia" : false,
	    "has_smp" : false,
	    "system" : [],
	    "memory" : [],
	    "cpu" : [],
	    "cdrom" : $[
		"manual" : [],
	    ],
	    "floppy" : $[
		"manual" : [],
	    ],
	],
    ];
    map I_WRITE = $[];
    map I_EXEC = $[
        "target" : $[
            "bash_output" : $[],
        ],
    ];
    TESTSUITE_INIT ([I_READ, I_WRITE, I_EXEC], 0);
    import "Printer";
    TESTSUITE_CLEAN ();

    Printer::printcap = [
	$[ "name" : "lp0", "uri" : "parallel:/dev/lp0",],
	$[ "name" : "lp1", "uri" : "parallel:/dev/lp1",],
	];
    Printer::selectPrinter (nil);
    DUMP (Printer::index);
    Printer::selectPrinter (-1);
    DUMP (Printer::index);
    Printer::selectPrinter (2);
    DUMP (Printer::index);
    Printer::selectPrinter (0);
    DUMP (Printer::index);
    Printer::selectPrinter (1);
    DUMP (Printer::index);
}
