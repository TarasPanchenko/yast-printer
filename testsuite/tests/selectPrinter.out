Dump	-1
Dump	-1
Dump	-1
Dump	0
Dump	1
